public/static/sdk/jsmr-0.1.js

```diff
@@ -145,22 +196,113 @@ var JsMr = Class.extend({
         );
     },
 
+    getData: function(task) {
+
+
+        var output = "";
+
+        var context = {
+            write: function (key, value) {
+                output = output.concat(key, ":", value, "\t");
+            },
+            state: {}
+        }
+
+        var _this = this;
+        var credentials = {accessKeyId: task.access_key, secretAccessKey: task.secret_key};
+        AWS.config.update(credentials);
+        //AWS.config.region = 'us-west-2';
+        var s3 = new AWS.S3();
+
+        //Fetch state from AWS
+         var job_id = task.job_id.toString();
+         var state_params = {
+             Bucket: task.bucket_name,
+             Key: job_id.concat('_state'),
+         }
+
+         s3.getObject(state_params, function (err, data) {
+         if (err){
+	     console.log("Error in fetching state");
+             console.log(err, err.stack); // an error occurred
+	 }
+         else {
+             // successful response
+             this.current_state = JSON.parse( data.Body.toString());
+             context.state = this.current_state;
+	     console.log("Fetched state is: ");
+	     console.log(this.current_state);
+         }
+         });
+	
+        //Fetch data from AWS
+        var params = {
+            Bucket: task.bucket_name,
+            Key: task.object_key,
+            Range: 'bytes='.concat(task.start_index, '-', task.end_index)
```

Comment:
I didn't use `bytes=` on the server, but my code is not tested. should I change mine to use `bytes=` ?